include/aspect/material_model/reaction_model/katz2003_mantle_melting.h

```diff
@@ -89,11 +89,47 @@ namespace aspect
                           const double pressure,
                           const double maximum_melt_fraction,
                           const NonlinearDependence::Dependence dependence) const;
+
+
+          /**
+           * Compute all the reaction rate variables needed for a reactive transport model based on the
+           * Katz 2003 formulation. Takes @p model inputs and @p model outputs to return.
+          */
+          void calculate_reaction_rate_outputs(const typename Interface<dim>::MaterialModelInputs &in,
+                                               typename Interface<dim>::MaterialModelOutputs &out) const;
```

Comment:
I still need the out variable to modify `out.reaction_terms` and `out.entropy_derivative_pressure` which I don't think are in `ReactionRateOutputs`. I could move `out.entropy_derivative_pressure` to `calculate_fluid_outputs` if you think it would fit there better, but I think `out.reaction_terms` needs to stay, so I'd need the full `out` object. I will move the viscosity outputs to `calculate_fluid_outputs` though.

Reply:
I'll defer to @jdannberg here, but if there is not away around passing `out` to this function, perhaps then no need to complicate further by my moving `out.entropy_derivative_pressure` to `calculate_fluid_outputs`?